### PR TITLE
Initial gtest integration

### PR DESCRIPTION
**Pull request by barath**
This branch adds gtest-based unit test support and includes a simple test of the time routines as a demonstration.

### CODE REVIEW

File: core/Makefile [hunk 1]
```diff
@@ -24,6 +24,11 @@ else
             Make sure $(abspath $(RTE_SDK)) is available and built)
 endif
 
+# gtest definitions
+GTEST_DIR = /usr/src/gtest
+CXXFLAGS += -isystem $(GTEST_DIR)/include
+CXXFLAGS += -g -Wall -pthread
```
Comment (sangjinhan) [inline core/Makefile]: This line is redundant..?

**Comment by sangjinhan** [comment]
Travis file needs to be updated too.

File: core/Makefile (hunk 1 above)
Comment (barath) [inline core/Makefile]: Good point, will remove.

**Comment by barath** [comment]
Updated pull request.

**Comment by sangjinhan** [comment]
Merged, thanks!

**Pull request merged**